src/v/bytes/iobuf.cc

```diff
@@ -200,3 +200,22 @@ bool iobuf::operator==(const iobuf& o) const {
     }
     return true;
 }
+
+bool iobuf::operator==(const std::string_view& o) const {
+    if (_size != o.size()) {
+        return false;
+    }
+    auto lhs_begin = byte_iterator(cbegin(), cend());
+    auto lhs_end = byte_iterator(cend(), cend());
+    auto c = o.data();
+    while (lhs_begin != lhs_end) {
```

Comment:
i don't see any tests for this. 

can you extract the iobuf changes into it's own patch set and simply depend on it. iobuf is pretty critical, so let's make sure you add the tests that verify that behavior independently of the coproc engine.